File: in_Python/0852 Peak Index in a Mountain Array.py
class Solution(object):
    def peakIndexInMountainArray(self, A):
        """
        :type A: List[int]
        :rtype: int
        """
        l, r = 0, len(A)-1
        while l < r:
            m = l + (r-l)/2
            if A[m] < A[m+1]:
                l = m+1
            else:
                r = m
        return l

File: in_Java/0852 Peak Index in a Mountain Array.java
class Solution {
    public int peakIndexInMountainArray(int[] A) {
        int m = Integer.MIN_VALUE;
        int res = -1;
        for(int i = 0; i < A.length; i++){
            if(A[i] > m){
                m = A[i];
                res = i;
            }
        }
        return res;
    }
}